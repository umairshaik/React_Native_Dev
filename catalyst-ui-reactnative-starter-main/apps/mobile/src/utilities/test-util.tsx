import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
