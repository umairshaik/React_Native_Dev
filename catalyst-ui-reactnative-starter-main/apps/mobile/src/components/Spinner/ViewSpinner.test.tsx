import {render} from '../../utilities/test-util';
import {ViewLoader} from './ViewSpinner';

describe('View Spinner', () => {
  test('render spinner component', () => {
    const component = render(<ViewLoader />);
    expect(component).toMatchSnapshot();
  });

  test('render spinner view', () => {
    const component = render(<ViewLoader />);
    expect(component.getByTestId('global-suspense-loader')).toBeTruthy();
  });
});
