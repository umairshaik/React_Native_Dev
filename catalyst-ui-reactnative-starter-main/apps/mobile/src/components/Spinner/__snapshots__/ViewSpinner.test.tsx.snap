// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Spinner render spinner component 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="global-suspense-loader"
>
  <ActivityIndicator
    color="black"
    size="large"
  />
</View>
`;
